[CI] DatafeedManagerTests testUpdateDatafeedFirstTimeMigrationShouldHonourExplicitProjectRoutingInUpdate failing

**Build Scans:**
- [elasticsearch-periodic #12412 / release-tests](https://gradle-enterprise.elastic.co/s/4avrpkh6gdo6m)
- [elasticsearch-periodic #12412 / release-tests](https://gradle-enterprise.elastic.co/s/5hddv6d2mxy7c)
- [elasticsearch-pull-request #155042 / checkpart4_release-tests](https://gradle-enterprise.elastic.co/s/twl6zp6b6wahq)
- [elasticsearch-pull-request #155042 / checkpart4_release-tests](https://gradle-enterprise.elastic.co/s/gjheajraa5lkm)
- [elasticsearch-pull-request #154992 / checkpart4_release-tests](https://gradle-enterprise.elastic.co/s/mgworax4d4v7y)
- [elasticsearch-pull-request #154992 / checkpart4_release-tests](https://gradle-enterprise.elastic.co/s/4to2onlup2xca)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ml:test" --tests "org.elasticsearch.xpack.ml.datafeed.DatafeedManagerTests.testUpdateDatafeedFirstTimeMigrationShouldHonourExplicitProjectRoutingInUpdate" -Dtests.seed=62FEBDFE5BA8A6C9 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=es-DO -Dtests.timezone=Poland -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ml.datafeed.DatafeedManagerTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testUpdateDatafeedFirstTimeMigrationShouldHonourExplicitProjectRoutingInUpdate'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: not null
     but: was null
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testUpdateDatafeedFirstTimeMigrationShouldHonourExplicitProjectRoutingInUpdate
- [main] 2 consecutive failures in step release-tests
- [main] 4 consecutive failures in step checkpart4_release-tests
- [main] 1 consecutive failures in pipeline elasticsearch-periodic
- [main] 6 failures in test testUpdateDatafeedFirstTimeMigrationShouldHonourExplicitProjectRoutingInUpdate (3.1% fail rate in 196 executions)
- [main] 2 failures in step release-tests (100.0% fail rate in 2 executions)
- [main] 4 failures in step checkpart4_release-tests (100.0% fail rate in 4 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.4% fail rate in 140 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -626,18 +626,6 @@ tests:
 - class: org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT
   method: test {yaml=scroll/10_basic/Basic scroll}
   issue: https://github.com/elastic/elasticsearch/issues/151698
-- class: org.elasticsearch.xpack.ml.datafeed.DatafeedManagerTests
-  method: testUpdateDatafeedFirstTimeMigrationShouldHonourExplicitProjectRoutingInUpdate
-  issue: https://github.com/elastic/elasticsearch/issues/151699
-- class: org.elasticsearch.xpack.ml.datafeed.DatafeedManagerTests
-  method: testUpdateDatafeedFirstTimeMigrationShouldDefaultProjectRoutingToOrigin
-  issue: https://github.com/elastic/elasticsearch/issues/151700
-- class: org.elasticsearch.xpack.ml.datafeed.DatafeedManagerTests
-  method: testUpdateDatafeedAlreadyMigratedShouldNotDefaultRoutingOnRekey
-  issue: https://github.com/elastic/elasticsearch/issues/151701
-- class: org.elasticsearch.xpack.ml.datafeed.DatafeedManagerTests
-  method: testUpdateDatafeedFirstTimeMigrationShouldNotOverrideExistingProjectRouting
-  issue: https://github.com/elastic/elasticsearch/issues/151702
 - class: org.elasticsearch.xpack.esql.CsvIT
   method: test {csv-spec:string.mvAppendNull}
   issue: https://github.com/elastic/elasticsearch/issues/151707

--- a/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/datafeed/DatafeedManagerTests.java
+++ b/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/datafeed/DatafeedManagerTests.java
@@ -1471,6 +1471,7 @@ public class DatafeedManagerTests extends ESTestCase {
 
     @SuppressWarnings("unchecked")
     public void testUpdateDatafeedFirstTimeMigrationShouldDefaultProjectRoutingToOrigin() {
+        assumeTrue("feature under test must be enabled", DatafeedConfig.DATAFEED_CROSS_PROJECT.isEnabled());
         Settings settings = Settings.builder().put("serverless.cross_project.enabled", true).put("xpack.security.enabled", false).build();
 
         DatafeedConfigProvider datafeedConfigProvider = mock(DatafeedConfigProvider.class);
@@ -1508,6 +1509,7 @@ public class DatafeedManagerTests extends ESTestCase {
 
     @SuppressWarnings("unchecked")
     public void testUpdateDatafeedFirstTimeMigrationShouldNotOverrideExistingProjectRouting() {
+        assumeTrue("feature under test must be enabled", DatafeedConfig.DATAFEED_CROSS_PROJECT.isEnabled());
         Settings settings = Settings.builder().put("serverless.cross_project.enabled", true).put("xpack.security.enabled", false).build();
 
         DatafeedConfigProvider datafeedConfigProvider = mock(DatafeedConfigProvider.class);
@@ -1552,6 +1554,7 @@ public class DatafeedManagerTests extends ESTestCase {
 
     @SuppressWarnings("unchecked")
     public void testUpdateDatafeedFirstTimeMigrationShouldHonourExplicitProjectRoutingInUpdate() {
+        assumeTrue("feature under test must be enabled", DatafeedConfig.DATAFEED_CROSS_PROJECT.isEnabled());
         Settings settings = Settings.builder().put("serverless.cross_project.enabled", true).put("xpack.security.enabled", false).build();
 
         DatafeedConfigProvider datafeedConfigProvider = mock(DatafeedConfigProvider.class);
@@ -1596,6 +1599,7 @@ public class DatafeedManagerTests extends ESTestCase {
 
     @SuppressWarnings("unchecked")
     public void testUpdateDatafeedAlreadyMigratedShouldNotDefaultRoutingOnRekey() {
+        assumeTrue("feature under test must be enabled", DatafeedConfig.DATAFEED_CROSS_PROJECT.isEnabled());
         Settings settings = Settings.builder().put("serverless.cross_project.enabled", true).put("xpack.security.enabled", false).build();
 
         DatafeedConfigProvider datafeedConfigProvider = mock(DatafeedConfigProvider.class);

